app/src/main/java/com/example/habitsmasher/ui/dashboard/EditHabitFragment.java

```diff
@@ -98,6 +112,110 @@ public void onClick(View v) {
+                if (_tracker.getMonday()){
+                    _tracker.setMonday(false);
+                }
+                //if monday wasn't already selected, select it
+                else{
+                    _tracker.setMonday(true);
+                }
+                Log.d("Tracker Status", _tracker.getDays());
+            }
+        });
+
+        tuesdayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(_tracker.getTuesday()){
+                    _tracker.setTuesday(false);
+                }
+                else{
+                    _tracker.setTuesday(true);
+                }
+                Log.d("Tracker Status", _tracker.getDays());
+            }
+        });
+
+        wednesdayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(_tracker.getWednesday()){
+                    _tracker.setWednesday(false);
+                }
+                else{
+                    _tracker.setWednesday(true);
+                }
+                Log.d("Tracker Status", _tracker.getDays());
+            }
+        });
+
+        thursdayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(_tracker.getThursday()){
+                    _tracker.setThursday(false);
+                }
+                else{
+                    _tracker.setThursday(true);
+                }
+                Log.d("Tracker Status", _tracker.getDays());
+            }
+        });
+
+        fridayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(_tracker.getFriday()){
+                    _tracker.setFriday(false);
+                }
+                else{
+                    _tracker.setFriday(true);
+                }
+                Log.d("Tracker Status", _tracker.getDays());
+            }
+        });
+
+        saturdayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(_tracker.getSaturday()){
+                    _tracker.setSaturday(false);
+                }
+                else{
+                    _tracker.setSaturday(true);
+                }
+                Log.d("Tracker Status", _tracker.getDays());
+            }
+        });
+
+        sundayButton.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if(_tracker.getSunday()){
+                    _tracker.setSunday(false);
+                }
+                else{
+                    _tracker.setSunday(true);
+                }
+                Log.d("Tracker Status", _tracker.getDays());
+            }
+        });
```

Comment:
This can also be extracted to its own method, and if it's the exact same is the other chunk of code above, you can maybe even call the same one.

Reply:
Extracted, in addition to making the buttons private. The extracted method also sets the initial states of the buttons too.